.github/workflows/unit_tests.yml

```
@@ -23,7 +23,12 @@ jobs:
       run: |
         go get gotest.tools/gotestsum@v0.4.2
         # TODO: validate bin/protoc-go.sh does not dirty the repo
-        gotestsum -- -cover -race -v -mod=readonly ./...
+        gotestsum -- -cover -coverprofile=coverage.out -race -v -mod=readonly ./...
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1
```

It's recommended to refer directly to the sha vs the tag (also please change in the other reference below):
```suggestion
      uses: codecov/codecov-action@29386c70ef20e286228c72b668a06fd0e8399192
```